app/models/survey.rb

```diff
@@ -17,9 +18,66 @@ class Survey < ActiveRecord::Base
 
   belongs_to :first_question, foreign_key: :first_question_id, class_name: 'Question'
 
-  validates :description, :name, presence: true
-
   def is_sendable?
-    return !first_question.nil?
+    first_question.present?
+  end
+
+  def generate_models
+    self.update!(
+      name: parameters['name'],
+      description: parameters['description'],
+    )
+
+    questions = generate_questions_and_response_choices(parameters['questions'])
+    self.update!(first_question: questions['1'])
```

Comment:
Shouldn't `'1'` be just `1`. Why are we accessing this hash as a string?

Edit: I just saw the tests they were changed from numbers to strings. What happened to the contract stating that these are numbers?